File: lib/Utils/ApiConfig.dart
const String apiBaseUrl = String.fromEnvironment(
  'API_BASE_URL',
  defaultValue: 'http://10.0.2.2:8080',
);

const String apiPrefix = String.fromEnvironment(
  'API_PREFIX',
  defaultValue: '/api/v1',
);

String buildApiBasePath() {
  final base = apiBaseUrl.endsWith('/')
      ? apiBaseUrl.substring(0, apiBaseUrl.length - 1)
      : apiBaseUrl;
  if (apiPrefix.isEmpty) return base;
  final prefix = apiPrefix.startsWith('/') ? apiPrefix : '/$apiPrefix';
  return '$base$prefix';
}

File: mobile_restaurant/utils/ApiConfig.ts
import { Platform } from 'react-native';

// Sử dụng IP LAN của máy tính để app trên điện thoại thật có thể gọi được API
// Kết nối qua dây cáp USB (adb reverse)
const DEFAULT_HOST = '127.0.0.1';

export const API_BASE_URL = `http://${DEFAULT_HOST}:8080`;
export const API_PREFIX = '/api/v1';

export function buildApiBasePath(): string {
  const base = API_BASE_URL.endsWith('/')
    ? API_BASE_URL.slice(0, -1)
    : API_BASE_URL;
  if (!API_PREFIX) return base;
  const prefix = API_PREFIX.startsWith('/') ? API_PREFIX : `/${API_PREFIX}`;
  return `${base}${prefix}`;
}

export function buildWsEndpoint(): string {
  const base = API_BASE_URL.endsWith('/')
    ? API_BASE_URL.slice(0, -1)
    : API_BASE_URL;
  return `${base}${API_PREFIX}/ws-chat`;
}
